### PR TITLE
[PEPPER-1103] [Playwright] Fix race condition with loading of Kit Search page

### PR DESCRIPTION
**Pull request by aweng98**
[PEPPER-1103](https://broadworkbench.atlassian.net/browse/PEPPER-1103)

[PEPPER-1103]: https://broadworkbench.atlassian.net/browse/PEPPER-1103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts [hunk 1]
```diff
@@ -22,26 +24,40 @@ export class ShippingSearchComponent implements OnInit {
...
     this.dsmService.getRealmsAllowed(Statics.SHIPPING).subscribe({
       next: data => {
         jsonData = data;
         jsonData.forEach((val) => {
-          this.allowedRealms.push(val);
+          if (sessionStorage.getItem(ComponentService.MENU_SELECTED_REALM) === val) {
+            allowedToSeeInformation = true;
+            this.allowedRealms.push(val);
+            this.disabled = false;
+          }
```
Comment (aweng98) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts:56]: Old code was not checking to see if selected realm is allowed to see Kit Search page.

Race condition issue is related to `getRealmsAllowed` request. If this request has not completed and user (playwright test is fast) has already filled out fields and clicked `Search Kit` button, error message `You are not allowed to see kit Information` will be shown.

File: ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts [hunk 2]
```diff
@@ -166,4 +182,8 @@ saveCompleted(): void{
     }
     return false;
   }
+
+  buttonDisabled(): boolean {
```
Comment (GiCharkviani) [inline ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts]: **OPTIONAL**
You could use here getter instead:
**TS:**
get buttonDisabled() {...}
**HTML:**
<element [disabled]="buttonDisabled" />

**Review by GiCharkviani — APPROVED** [review]
LGTM

File: ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts (hunk 2 above)
Comment (aweng98) [reply ddp-workspace/projects/ddp-dsm-ui/src/app/shipping-search/shipping-search.component.ts]: 👍🏻 Good idea!

**Pull request merged**